src/components/implementation/no_interface/llbooter/llbooter.c

```diff
@@ -299,6 +320,64 @@ comps_init(void)
 	return;
 }
 
+void 
+chkpt_comp_init(struct crt_comp *comp, struct crt_chkpt *chkpt, char *name)
+{
+#ifdef ENABLE_CHKPT
+	/* create the component */
+	void *elf_hdr;
+	int   keylen;
+	compid_t id = crt_ncomp() + 1;
+	const char *root = "binaries/";
+	int   len  = strlen(root);
+	char  path[INITARGS_MAX_PATHNAME];
+
+	assert(id < MAX_NUM_COMPS && id > 0 && name);
+	memset(path, 0, INITARGS_MAX_PATHNAME);
+	strncat(path, root, len);
+	assert(path[len] == '\0');
+	strncat(path, name, INITARGS_MAX_PATHNAME - len);
+	assert(path[INITARGS_MAX_PATHNAME - 1] == '\0'); /* no truncation allowed */
+
+	if (id == cos_compid()) {
+		/* this should never happen */
+		assert(0);
+	} else {
+		if (crt_comp_create_from(comp, name, id, chkpt)) {
+			printc("Error constructing the resource tables and image of component %s.\n", comp->name);
+			BUG();
+		}
+	}
+	assert(comp->refcnt != 0);
+
+	/* TODO: create the thread/execution context */
+	struct crt_comp_exec_context ctxt = { 0 };
+	/* TODO: assume only chkpt components that can be executed, but aren't schedulers */
```

Comment:
I don't really parse this comment. Can it be reworded? Am I just being obtuse?

Reply:
I removed the first one because agreed that it doesn't make sense. 

I will re-phrase the second one -- it's left from a discussion about not checkpointing schedulers/capability managers, so we aren't copying the capability tables (but yeah, the way it's phrased with that code doesn't make sense)